rinja_derive/src/generator.rs

```diff
@@ -1364,12 +1364,14 @@ impl<'a> Generator<'a> {
             ));
         }
 
-        if args.len() != 1 {
-            return Err(ctx.generate_error("unexpected argument(s) in `json` filter", node));
-        }
         buf.write(CRATE);
         buf.write("::filters::json(");
         self._visit_args(ctx, buf, args)?;
+        match args.len() {
+            1 => buf.write(", ()"),
```

Comment:
Instead of adding a trait to handle integer/void, why not write a specialized json filter called `json_empty`?

Second question: why not forcing the type of `indent` to be a `usize` so we don't have to handle types and other things? I find it a bit strange to allow users to pass a negative number of even `None` as arguments.